core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -242,8 +243,13 @@ public ContextAwareScheduledExecutorService blockingTaskExecutor() {
             return blockingTaskExecutor;
         }
 
-        return blockingTaskExecutor = ContextAwareScheduledExecutorService.of(
-                this, config().server().config().blockingTaskExecutor());
+        final ScheduledExecutorService executor;
+        if (config().blockingTaskExecutor() != null) {
+            executor = config().blockingTaskExecutor();
```

Comment:
Could we make `config().blockingTaskExecutor()` never return `null`?

Reply:
> Could we make `config().blockingTaskExecutor()` never return `null`?

Yes, I did this at first and later realize that if we do that, users won't be able to distinguish server-level / route-level blockingTaskExecutor, is it OK? If so I'll do this

Reply:
It's OK. We already do that for other per-route properties. Please feel free to let me know if you have any use case for distinguishing server-level and route-level property.